rbd/group_snap.go

```diff
@@ -2,14 +2,26 @@ package rbd
 
 /*
 #cgo LDFLAGS: -lrbd
+#include <errno.h>
 #include <stdlib.h>
 #include <rbd/librbd.h>
+
+extern int snapRollbackCallback(uint64_t, uint64_t, uintptr_t);
+
+// inline wrapper to cast uintptr_t to void*
+static inline int wrap_rbd_group_snap_rollback_with_progress(
+		rados_ioctx_t group_p, const char *group_name,
+		const char *snap_name, uintptr_t arg) {
+	return rbd_group_snap_rollback_with_progress(
+		group_p, group_name, snap_name, (void*)snapRollbackCallback, (void*)arg);
```

Comment:
you could typecast to `(librbd_progress_fn_t*)` instead of `(void*)` to be more strict (as Sven mentioned in the syncup call)

Reply:
Changed. Thanks!